load.go

```diff
@@ -335,8 +342,30 @@ func (c *Command) processBuildArgs(lex *shell.Lex, args map[string]string, name
 	return nil
 }
 
+func (s Spec) FillDefaults() Spec {
+	for _, src := range s.Sources {
```

Comment:
I think this is not doing what we want it to since `Sources` is `map[string]Source` and not `map[string]*Source`.

To make this work we either need to do something like:
```suggestion
	for i, src := range s.Sources {
		s := &src
		fillDefaults(&s)
		s.Sources[i] = *s
	}
```

Or I think we can just change `Sources` to be a `

Reply:
Also seems like we must be missing a test for this.

Reply:
Good catch, thank you! I believe the original code was putting the filled source back in the source map as you have it here, but I must have forgotten that line. We should definitely add a test that would catch this